### ISSUE TITLE
Error in array remove and add operation, Maximum call stack size exceeded

### ISSUE THREAD

**Issue by dev-kanishk** [ngx-formly/ngx-formly#3794]
**Description**
For the following json schema, When removing an element from the array, I am getting the error "Maximum call stack size exceeded". And further all the add operation are also giving same error.

```
{
        type: 'object',
        properties: {
          heading: {
            title: 'Heading',
            type: 'array',
            items: {
              oneOf: [
                {
                  title: 'Option 1: Standard heading',
                  type: 'string',
                  oneOf: [
                    { const: 'Heading 1' },
                    { const: 'Heading 2' },
                    { const: 'Heading 3' },
                  ],
                },
                {
                  title: 'Option 2: Custom heading',
                  type: 'string',
                  default: 'def',
                },
              ],
            },
          },
        },
      }
```

**Minimal Reproduction**
https://stackblitz.com/edit/angular-kuxmzz?file=src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fapp.component.html,src%2Fapp%2Farray.type.ts

**Your Environment**

- Formly version: 6.2.1

**Additional context**
Support for the oneOf schema in array was added in this commit: https://github.com/ngx-formly/ngx-formly/commit/bcf930b8f810b5dcff5607f5b58805186b0298e8

which fixed: #3769, fix #3785

**Comment by dev-kanishk** [comment]
Hey @aitboudad, I found that in remove operation we are resetting the keys, but for supporting the oneOf as array element we set the element keys to null. In this fix we are navigating to the correct field and then resetting the keys.

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.2.2](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.2.2) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
